File: Day 8/8th-day.java
/*
Task
Given n names and phone numbers, assemble a phone book that maps friends' names to their respective phone numbers. You will
then be given an unknown number of names to query your phone book for. For each name queried, print the associated entry from
your phone book on a new line in the form name=phoneNumber: if an entry for name is not found, print Not found instead.
Note: Your phone book should be a Dictionary/Map/HashMap data structure.
*/


import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine(); // Consume the newline character after reading integer input

        Map<String, String> phoneBook = new HashMap<>();

        // Populate the phone book
        for (int i = 0; i < n; i++) {
            String line = scanner.nextLine();
            String[] parts = line.split(" ");
            String name = parts[0];
            String phoneNumber = parts[1];
            phoneBook.put(name, phoneNumber);
        }

        // Query the phone book
        while (scanner.hasNextLine()) {
            String query = scanner.nextLine().trim();
            if (phoneBook.containsKey(query)) {
                System.out.println(query + "=" + phoneBook.get(query));
            } else {
                System.out.println("Not found");
            }
        }

        scanner.close();
    }
}

File: Day 8/8th-day.cpp
/*
Task
Given n names and phone numbers, assemble a phone book that maps friends' names to their respective phone numbers. You will
then be given an unknown number of names to query your phone book for. For each name queried, print the associated entry from
your phone book on a new line in the form name=phoneNumber: if an entry for name is not found, print Not found instead.
Note: Your phone book should be a Dictionary/Map/HashMap data structure.
*/

#include <iostream>
#include <sstream>
#include <unordered_map>

using namespace std;

int main() {
    int n;
    cin >> n;
    cin.ignore(); // Ignore the newline character after reading integer input

    unordered_map<string, string> phoneBook;

    // Populate the phone book
    for (int i = 0; i < n; ++i) {
        string line;
        getline(cin, line);
        istringstream iss(line);
        string name, phoneNumber;
        iss >> name >> phoneNumber;
        phoneBook[name] = phoneNumber;
    }

    // Query the phone book
    string query;
    while (getline(cin, query)) {
        if (query.empty()) break;
        if (phoneBook.find(query) != phoneBook.end()) {
            cout << query << "=" << phoneBook[query] << endl;
        } else {
            cout << "Not found" << endl;
        }
    }

    return 0;
}


